### ISSUE TITLE
Scrolling of inside scrollable element results in skipping

### ISSUE THREAD

**Issue by vualeks** [MarioIannotta/PullUpController#17]
Hi,

I've just tried the example from the project, added just a few more elements to tableview that is inside and got this behaviour that I've recorded in video https://youtu.be/q4cU4kW-rcQ
It happens when panel is fully open, inside tableview is scrolled a bit up and then when you pulling it down it starts skipping when it reaches the end by pulling the panel to compensate. Maybe it's not well explained in words, but you can see it in video :)

Regards,

Aleksandar

**Comment by obianachebe** [comment]
I am seeing this issue as well

**Comment by MarioIannotta** [comment]
The bug is inside the function `handleInternalScrollViewPanGestureRecognizer(_ gestureRecognizer: UIPanGestureRecognizer)`.

The problem is that when scrolling down from the beginning of the list `yTranslation` (`handlePanGestureRecognizer#203`) is 0,1,2,3 and so on; that means `topConstraint.constant` is progressively updated.

If the scroll starts, for example, from the second row (height = 30), `yTranslation` will be 0,1,2 but [1]`shouldScrollingDownTriggerGestureRecognizer || shouldScrollingUpTriggerGestureRecognizer` won't evaluate true untill the scroll view is scrolled to the end.

When [1] will be evaluated true `yTranslation` will be `30` and `topConstraint.constant` jumps from "some value" to "some value" + 30.

**Comment by idanboa** [comment]
any progress on this issue? @MarioIannotta

**Comment by MarioIannotta** [comment]
@vualeks @obianachebe @idanboa please checkout [`bugfix/17`](https://github.com/MarioIannotta/PullUpController/tree/bugfix/17) and let me know if the fix works for you too and if you notice any regressions.
Thank you everyone!

**Comment by vualeks** [comment]
Hey @MarioIannotta ,

I've just tried it and it's much better than before, but still not quite as that google maps pull up. It doesn't constantly jump now when you pull it in that problematic way ( :D ), but pull view jumps only once to align to the scrollview position and then if you start pushing up (backwards) it jumps once again to realign itself. If you couldn't understand what I tried, I can make a video. Anyway, for me it's good enough, but you decide if you want to pursue any further. 

Thanks for your work,

Aleksandar

**Comment by MarioIannotta** [comment]
@vualeks I got your point and you're right, this is an little improvement but the scrolling is not smooth and natural as it should be. 
The "fix" has been merged in 3ef7316c9ced420381828e2692e3ffa390e04a63 (if you're using CocoaPods, please run a `pod update`) but I'll keep this issue open hoping that in future we come out with a better solution.

**Comment by MarioIannotta** [comment]
Ok I think I've got the final solution now.
@vualeks please checkout again [`bugfix/17`](https://github.com/MarioIannotta/PullUpController/tree/bugfix/17) and let me know your opinion.
Thanks!

**Comment by vualeks** [comment]
Yeah, that's it, now that part works ok. I've noticed one more thing now, not sure if it also was on previous try, but I'm pretty sure it isn't present on release version. When you pull down the inside list quickly and release the tap before you've reached the end of the table view list, it will continue to scroll inertially but when it reaches the end it will also pull the view down, essentially close it, and that should not happen. See if you can replicate it.

Aleksandar

EDIT: I'm not sure that it is the reason that the list reaches the end, but try it few times, sometimes you get the pull down event even when the list has not reached the end. I'm only trying on simulator here, don't have an iphone to try, maybe only related to sim.

**Comment by MarioIannotta** [comment]
With 00f3494dc9ca86705327fc4f70716e6bceb0e5ff the inertial scroll should be ok.

**Comment by vualeks** [comment]
Yes, now it works perfectly. I had one more idea for an improvement, but I'll open another issue for that some other time when I test if more thoroughly.

**Comment by idanboa** [comment]
great fix @MarioIannotta ! thanks

**Comment by MarioIannotta** [comment]
Merged with 00f3494dc9ca86705327fc4f70716e6bceb0e5ff and available through CocoaPods with the version `0.4.1`.
Thanks everyone for the support 🍻
